framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
That ```historyControl()``` wouldn't happen if it were done that way.  Based on the comment above that call, it sounds like we'd need to call a method on the predictor at the beginning of the time step to do what you suggest (which probably isn't terribly difficult).  Is that what we should do?